test/cppsuite/test_harness/workload_validation.h

```diff
@@ -0,0 +1,395 @@
+            debug_info("Operation type is " + std::to_string(value_operation_type), _trace_level,
+              DEBUG_INFO);
+
+            if (static_cast<tracking_operation>(value_operation_type) ==
+              tracking_operation::CREATE) {
+                created_collections.push_back(key_collection_name);
+            } else if (static_cast<tracking_operation>(value_operation_type) ==
+              tracking_operation::DELETE_COLLECTION) {
+                created_collections.erase(std::remove(created_collections.begin(),
+                                            created_collections.end(), key_collection_name),
+                  created_collections.end());
+            }
+        }
+
+        return created_collections;
+    }
+
+    /*
+     * Parse the tracked operations to build a representation in memory of the collections at the
+     * end of the test. tracking_collection_name is the tracking collection used to save the
+     * operations performed on the collections during the test. collection_name is the collection
+     * that needs to be represented in memory.
+     */
+    void
+    parse_collections_operations(WT_SESSION *session, const std::string &tracking_collection_name,
+      const std::string &collection_name,
+      std::map<std::string, std::map<int, std::string *> *> &collections)
+    {
+        WT_CURSOR *cursor;
+        int exact;
+        int error_code;
+        const char *key_collection_name, *value;
+        int key, key_timestamp, value_operation_type;
+
+        testutil_check(
+          session->open_cursor(session, tracking_collection_name.c_str(), NULL, NULL, &cursor));
+
+        /* Our keys start at 0. */
+        cursor->set_key(cursor, collection_name.c_str(), 0);
+        error_code = cursor->search_near(cursor, &exact);
+
+        while (error_code == 0) {
+            testutil_check(cursor->get_key(cursor, &key_collection_name, &key, &key_timestamp));
+            testutil_check(cursor->get_value(cursor, &value_operation_type, &value));
+
+            debug_info(
+              "Collection name is " + std::string(key_collection_name), _trace_level, DEBUG_INFO);
+            debug_info("Key is " + std::to_string(key), _trace_level, DEBUG_INFO);
+            debug_info("Timestamp is " + std::to_string(key_timestamp), _trace_level, DEBUG_INFO);
+            debug_info("Operation type is " + std::to_string(value_operation_type), _trace_level,
+              DEBUG_INFO);
+            debug_info("Value is " + std::string(value), _trace_level, DEBUG_INFO);
+
+            if (std::string(key_collection_name) != collection_name)
+                break;
+
+            /* Replay the current operation. */
+            switch (static_cast<tracking_operation>(value_operation_type)) {
+
+                break;
+            case tracking_operation::DELETE_KEY: {
+                /* Check if the collection exists. */
+                if (collections.count(key_collection_name) < 1)
```

Comment:
I'm confused by this, why are we checking if the collection here exists?

Reply:
Collections are created in memory thanks to the schema_tracking_table. If for some reason we miss the CREATE operation for this collection, it will not be allocated in memory, then we cannot interact with it at all using the operation_tracking_table.

Reply:
Lets talk about this function in person.